src/owr/ObolErc1155Recipient.sol

```diff
@@ -77,199 +88,141 @@ contract ObolErc1155Recipient is ERC1155, Ownable, IERC1155Receiver {
+    
+    // mint to sender
+    _mint(msg.sender, mintedId, 1, "");
+    ownerOf[mintedId] = msg.sender;
 
-    // increase supply
-    tokenInfo[id].maxSupply += amount;
+    emit Minted(_partitionId, mintedId, msg.sender);
   }
 
   /// @notice decreases totalSupply for token id
-  /// @param id token id
-  /// @param amount newly removed supply
-  function burn(uint256 id, uint256 amount) external {
+  /// @param _tokenId token id
+  function burn(uint256 _tokenId) external {
     // validate
-    if (!isReceiverOf(id)) revert InvalidOwner();
-    if (amount == 0) revert InvalidTokenAmount();
+    if (!isOwnerOf(_tokenId)) revert InvalidOwner();
+
+    // retrieve OWR
+    address _owr = partitions[tokensPartition[_tokenId]].owr;
+    if (_owr == address(0)) revert OwrNotValid();
 
-    uint256 minEthAmount = MIN_ETH_EXIT_AMOUNT  * amount;
     uint256 ethBalanceBefore = address(this).balance;
-    IOptimisticWithdrawalRecipient(tokenInfo[id].owr).distributeFunds();
+    IOptimisticWithdrawalRecipient(_owr).distributeFunds();
     uint256 ethBalanceAfter = address(this).balance;
     uint256 ethReceived = ethBalanceAfter - ethBalanceBefore;
+    // TODO: check `ethReceived` amount
 
-    if(ethReceived < minEthAmount) revert InvalidBurnAmount(minEthAmount, ethReceived);
+    if(ethReceived < MIN_ETH_EXIT_AMOUNT) revert InvalidBurnAmount(MIN_ETH_EXIT_AMOUNT, ethReceived);
 
-    _burn(msg.sender, id, amount);
+    _burn(msg.sender, _tokenId, 1);
 
-    totalSupply[id] -= amount;
-    totalSupplyAll -= amount;
+    totalSupply[_tokenId]--;
+    totalSupplyAll--;
 
-    (bool sent,) = tokenInfo[id].receiver.call{value: ethReceived}("");
+    (bool sent,) = msg.sender.call{value: ethReceived}("");
     if (!sent) revert TransferFailed();
   }
 
+  /// @notice triggers `OWR.distributeFunds` and updates claimable balances for partition
+  /// @param _tokenId token id
+  /// @param _splitConfig pull split configuration
+  function updateRewards(uint256 _tokenId, address _distributor, IPullSplit.PullSplitConfiguration calldata _splitConfig) external {
+    uint256 _partitionId = tokensPartition[_tokenId];
+    address _owr = partitions[tokensPartition[_tokenId]].owr;
+    if (_owr == address(0)) revert OwrNotValid();
+
+
+    // call .distribute() on OWR
+    uint256 balanceBefore = _getTokenBalance(_owr);
+    IOptimisticWithdrawalRecipient(_owr).distributeFunds();
+    _distributeSplitsRewards(_owr, _distributor, _splitConfig);
+    uint256 balanceAfter = _getTokenBalance(_owr);
+
+    uint256 _totalClaimable = balanceAfter - balanceBefore;
+    if (_totalClaimable > 0) {
+      uint256 count = partitionTokens[_partitionId].length;
+      uint256 _reward = _totalClaimable / count;
+      for(uint i; i < count; i++) {
+        address _owner = ownerOf[partitionTokens[_partitionId][i]];
+        claimable[_owner] += _reward;
+      }
+    }
+  } 
+
+  /// @notice claim rewards
+  function claim() external {
```

Comment:
we should pass the address as a param i.e. `function claim(address holder) {}`

This would allow us to batch call `claim()` & `distributeRewards()` when needed